### DISCUSSION TITLE
<Style> doesn't work with extended controls

### DISCUSSION THREAD

**Discussion by duck-dev** [AvaloniaUI/Avalonia#9419] [Q&A] [answered]
**Describe the bug**
I'm trying to extend the `Button` control, so the `PointerPressed` event can be invoked through a left-click as well. Setting the `PointerPressed` property of a normal `Button` results in a detection of a right-click only. I can't add a handler manually, because the `Button` element is an element listed inside of an `ItemsControl`. That's why I decided to extend the `Button` control with a custom control.

Now, the button's are being displayed and the properties that are directly being set on the control's declaration are applied, but everything inside the external style inclusion (`YearListViewStyle.axaml` in the example below) is simply being ignored. 

**To Reproduce**

`CustomButton.cs`:
```cs
public sealed class CustomButton : Button, IStyleable
{
    public event EventHandler<PointerPressedEventArgs> PointerPressedPreview = delegate {  };
        
    Type IStyleable.StyleKey => typeof(ContentControl);

    protected override void OnPointerPressed(PointerPressedEventArgs args)
    {
        PointerPressedPreview(this, args);
        base.OnPointerPressed(args);
    }
}
```

Some UserControl's content:
```xaml
<UserControl.Styles>
    <StyleInclude Source="/src/Styles/YearListViewStyle.axaml"/>
</UserControl.Styles>

<ScrollViewer VerticalScrollBarVisibility="Auto"
              HorizontalScrollBarVisibility="Disabled">
    <ItemsControl Items="{Binding Items}">
        <ItemsControl.ItemsPanel>
            <ItemsPanelTemplate>
                <WrapPanel Orientation="Horizontal"/>
            </ItemsPanelTemplate>
         </ItemsControl.ItemsPanel>
         <ItemsControl.ItemTemplate>
             <DataTemplate>
                 <customControls:CustomButton Command="{Binding $parent[Window].DataContext.OpenYear}"
                                              CommandParameter="{Binding}"
                                              Classes="YearButton"
                                              Width="500"
                                              Height="300"
                                              Background="#c7cad1"
                                              Margin="30, 30"
                                              BorderThickness="0"
                                              DragDrop.AllowDrop="True"
                                              PointerPressedPreview="BeginDrag">
                     <!-- Some content inside the button -->
                 </customControls:CustomButton>
             </DataTemplate>
         </ItemsControl.ItemTemplate>
    </ItemsControl>
</ScrollViewer>
```

`YearListViewStyle.axaml`:
```xaml
<Styles xmlns="https://github.com/avaloniaui"
        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
        xmlns:customControls="clr-namespace:GradeManagement.Views.CustomControls">
    
    <Style Selector="customControls|CustomButton.year-button">
        <Setter Property="RenderTransform" Value="scale(1)"></Setter>
    </Style>
    <Style Selector="customControls|CustomButton.year-button /template/ ContentPresenter">
        <Setter Property="CornerRadius" Value="10"></Setter>
    </Style>
    <Style Selector="customControls|CustomButton.year-button:pointerover">
        <Setter Property="Cursor" Value="Hand"></Setter>
        <Setter Property="RenderTransform" Value="scale(0.95)"></Setter>
    </Style>
    <Style Selector="customControls|CustomButton.year-button:pressed /template/ ContentPresenter">
        <Setter Property="Background" Value="#b8babf"></Setter>
    </Style>
</Styles>
```

Everything that's inside this `YearListViewStyle.axaml` is **NOT** being applied to the custom button. It works fine with a normal button, but as I described above, I have to use a custom button, unless there is a better solution I'm not aware of.

**Expected behavior**
The properties modified with the <Setter> tags inside the <Style> tags of an externally included style should be applied to the extended control.

**Desktop (please complete the following information):**
- OS: Windows 10 21H1
- Version: 0.10.3

**Comment by maxkatz6** [comment, marked as answer]
"customControls|CustomButton" won't work, because you explicitly told style-subsystem, that this control should use "ContentControl" type:
```
Type IStyleable.StyleKey => typeof(ContentControl);
```
This is expected behavior, but we always look for fitting improvements to make it easier.
There was relevant discussion not so long ago https://github.com/AvaloniaUI/Avalonia/discussions/6353
And proposal https://github.com/AvaloniaUI/Avalonia/issues/6382

**Comment by maxkatz6** [comment]
But specifically in your case you probably don't need StyleKey at all.
ContentControl styles are very simple and you can copy them (as it is usually done for ContentControl-like control, such as simple Button) - https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Themes.Fluent/Controls/ContentControl.xaml

**Comment by duck-dev** [comment]
Hello @maxkatz6 .

I had the misfortune that my PC broke almost immediately after formulating the issue. Unfortunately, I didn't have time to get the PC working again after that, until now. Therefore the extremely delayed answer, apologies for that.

So, back to the issue. It's quite a long time since I asked this question and I'm not 100% sure what I could do to solve the problem now. How could I still use the styling in the separate file with the custom control, designed as above? I changed the type from `ContentControl` to `Button`, which didn't change anything at all. I'd be happy if you could make it a bit clearer to me, thank you very much.

**Comment by timunie** [comment]
@duck-dev 

I think I see two issues: 

- `Type IStyleable.StyleKey => typeof(ContentControl);`  should probably be `Type IStyleable.StyleKey => typeof(Button);` ?
- Your Selector is `Selector="customControls|CustomButton.year-button"` but your class you request is `YearButton`. Please try to correct either one.

**Comment by timunie** [comment]
I've converted this issue into Q&A as it's not a bug. If you disagree please let me know.

**Comment by Shadowblitz16** [comment]
@timunie This needs to be fixed for Styles.Design.PreviewWith
